src/App.tsx

```diff
@@ -1,26 +1,139 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todosService from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorsEnum } from './utils/ErrorsEnum';
+import { Error } from './components/Error/Error';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todoList, setTodosList] = useState<Todo[]>([]);
+  const [filteredTodoList, setFilteredTodoList] = useState<Todo[]>([]);
```

Comment:
You don't have to save _filteredTodoList_ in your state.
Instead you can use an _useMemo_ hook to create a variable outside of the state that filters the result based on other state aspects.
The way your code is working, you have one state handling the filter type, another handling the filter result, when in fact, whenever the filter type changes, the filter result will also change, so they can be used in a single state.